File: capacitorbase/android/src/main/java/com/viewnext/capacitorbase/BasePlugin.java
package com.viewnext.capacitorbase;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@SuppressWarnings("unused")
@CapacitorPlugin(name = "BasePlugin")
public class BasePlugin extends Plugin {
  private final Base implementation = new Base();

  @PluginMethod
  public void echo(PluginCall call) {
    String value = call.getString("value");
    JSObject ret = new JSObject();
    ret.put("value", implementation.echo(value));
    call.resolve(ret);
  }
}

File: capacitorbase/ios/Plugin/BasePlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(BasePlugin)
public class BasePlugin: CAPPlugin {
    private let implementation = Base()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
